src/minters/signature/ZoraCreatorSignatureMinterStrategy.sol

```diff
@@ -0,0 +1,248 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.17;
+
+import {Enjoy} from "_imagine/mint/Enjoy.sol";
+import {MerkleProof} from "@openzeppelin/contracts/utils/cryptography/MerkleProof.sol";
+import {IMinter1155} from "../../interfaces/IMinter1155.sol";
+import {ICreatorCommands} from "../../interfaces/ICreatorCommands.sol";
+import {SaleStrategy} from "../SaleStrategy.sol";
+import {ICreatorCommands} from "../../interfaces/ICreatorCommands.sol";
+import {SaleCommandHelper} from "../utils/SaleCommandHelper.sol";
+import {LimitedMintPerAddress} from "../utils/LimitedMintPerAddress.sol";
+import {EIP712} from "@openzeppelin/contracts/utils/cryptography/EIP712.sol";
+import {ECDSA} from "@openzeppelin/contracts/utils/cryptography/ECDSA.sol";
+import {IReadableAuthRegistry} from "../../interfaces/IAuthRegistry.sol";
+
+/*
+
+
+             ░░░░░░░░░░░░░░              
+        ░░▒▒░░░░░░░░░░░░░░░░░░░░        
+      ░░▒▒▒▒░░░░░░░░░░░░░░░░░░░░░░      
+    ░░▒▒▒▒░░░░░░░░░░░░░░    ░░░░░░░░    
+   ░▓▓▒▒▒▒░░░░░░░░░░░░        ░░░░░░░    
+  ░▓▓▓▒▒▒▒░░░░░░░░░░░░        ░░░░░░░░  
+  ░▓▓▓▒▒▒▒░░░░░░░░░░░░░░    ░░░░░░░░░░  
+  ░▓▓▓▒▒▒▒▒▒░░░░░░░░░░░░░░░░░░░░░░░░░░░  
+  ░▓▓▓▓▓▒▒▒▒░░░░░░░░░░░░░░░░░░░░░░░░░░  
+   ░▓▓▓▓▒▒▒▒▒▒░░░░░░░░░░░░░░░░░░░░░░░  
+    ░░▓▓▓▓▒▒▒▒▒▒░░░░░░░░░░░░░░░░░░░░    
+    ░░▓▓▓▓▓▓▒▒▒▒▒▒▒▒░░░░░░░░░▒▒▒▒▒░░    
+      ░░▓▓▓▓▓▓▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒░░      
+          ░░▓▓▓▓▓▓▓▓▓▓▓▓▒▒░░░          
+
+               OURS TRULY,
+
+
+    github.com/ourzora/zora-1155-contracts
+
+ */
+
+/// @title ZoraCreatorSignatureMinterStrategy
+/// @notice Mints tokens based on signature created by an authorized signer
+/// @author @oveddan
+contract ZoraCreatorSignatureMinterStrategy is Enjoy, SaleStrategy, LimitedMintPerAddress, EIP712 {
+    using SaleCommandHelper for ICreatorCommands.CommandSet;
+
+    /// @notice General signatue sale settings
+    struct SalesConfig {
+        /// @notice Registry that decides which account is authorized to sign signatures
+        /// to mint for the sale
+        IReadableAuthRegistry authorizedSignatureCreators;
+    }
+
+    // target -> settings
+    mapping(address => SalesConfig) signatureSaleSettings;
+    // target contract -> unique nonce -> if has been minted already
+    mapping(address => mapping(bytes32 => bool)) private minted;
```

Comment:
looking at [this tweet](https://twitter.com/danielvf/status/1675825631240519680), it's great we are **not** doing signature used by signature hash.